File: claude/node/securable/src/models/PasswordResetToken.js
'use strict';

const { DataTypes } = require('sequelize');

/**
 * Password reset tokens.
 * Token value stored as bcrypt hash — raw token sent by email only (Confidentiality).
 * Expiry enforced at query time, not by database TTL.
 */
module.exports = (sequelize) => {
  const PasswordResetToken = sequelize.define(
    'PasswordResetToken',
    {
      id: {
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
        primaryKey: true,
      },
      userId: {
        type: DataTypes.UUID,
        allowNull: false,
        references: { model: 'users', key: 'id' },
      },
      tokenHash: {
        type: DataTypes.STRING(255),
        allowNull: false,
      },
      expiresAt: {
        type: DataTypes.DATE,
        allowNull: false,
      },
      usedAt: {
        type: DataTypes.DATE,
        allowNull: true,
      },
    },
    {
      tableName: 'password_reset_tokens',
      indexes: [{ fields: ['user_id'] }],
    }
  );

  PasswordResetToken.associate = (models) => {
    PasswordResetToken.belongsTo(models.User, {
      foreignKey: 'userId',
      as: 'user',
    });
  };

  return PasswordResetToken;
};

File: claude/aspnet/securable/LooseNotes/Models/PasswordResetToken.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LooseNotes.Models;

public class PasswordResetToken
{
    public int Id { get; set; }

    // Stored as hash — raw token never persisted
    [Required, MaxLength(256)]
    public string TokenHash { get; set; } = string.Empty;

    [Required]
    public string UserId { get; set; } = string.Empty;

    [ForeignKey(nameof(UserId))]
    public ApplicationUser? User { get; set; }

    public DateTime ExpiresAt { get; set; }
    public bool IsUsed { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
